lib/promise.rb

```diff
@@ -25,11 +25,18 @@ def self.all(enumerable)
     Group.new(new, enumerable).promise
   end
 
-  def self.map_value(obj)
-    if obj.is_a?(Promise)
-      obj.then { |value| yield value }
-    else
-      yield obj
+  if Gem.ruby_version >= Gem::Version.new('2.6.0')
+    # @deprecated Use {Object#then} instead
+    def self.map_value(obj, &block)
+      obj.then(&block)
+    end
```

Comment:
When no block is given, `Object#then` might return an `Enumerator`. Is it an expected behavior? 🤔

Reply:
No, I didn't consider that.  It is probably harmless, but I've changed it back to `obj.then { |value| yield value }` to keep it consistent with the existing behaviour.

Reply:
Sound reasonable 😄 👍